src/network/networkd-link-bus.c

```diff
@@ -195,9 +195,16 @@ int bus_link_method_set_domains(sd_bus_message *message, void *userdata, sd_bus_
         if (r < 0)
                 return r;
 
+        search_domains = ordered_set_new(&string_hash_ops_free);
+        if (!search_domains)
+                return -ENOMEM;
+
+        route_domains = ordered_set_new(&string_hash_ops_free);
+        if (!route_domains)
+                return -ENOMEM;
```

Comment:
- `link->route_domains == NULL`: static route domains, that is, `link->network->route_domains` will be used,
- `ordered_set_isempty(link->route_domains)`: the DBus method was called without route domains,
- otherwise: the DBus method was called with valid route domains.

Previously, if the DBus method was called without route domains, e.g. all domains are search domains, then `link->route_domains` will be NULL, and we cannot distinguish the first two cases in the above.